Balint-list1.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python -tt
+# Copyright 2010 Google Inc.
+# Licensed under the Apache License, Version 2.0
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Google's Python Class
+# http://code.google.com/edu/languages/google-python-class/
+
+# Basic list exercises
+# Fill in the code for the functions below. main() is already set up
+# to call the functions with a few different inputs,
+# printing 'OK' when each function is correct.
+# The starter code for each function includes a 'return'
+# which is just a placeholder for your code.
+# It's ok if you do not complete all the functions, and there
+# are some additional functions to try in list2.py.
+
+# A. match_ends
+# Given a list of strings, return the count of the number of
+# strings where the string length is 2 or more and the first
+# and last chars of the string are the same.
+# Note: python does not have a ++ operator, but += works.
+def match_ends(words):
+    count = 0
+    for item in words:
+        if len(item)>=2 and item[0] == item[-1]:
+            count += 1
+    return count
+
+# B. front_x
+# Given a list of strings, return a list with the strings
+# in sorted order, except group all the strings that begin with 'x' first.
+# e.g. ['mix', 'xyz', 'apple', 'xanadu', 'aardvark'] yields
+# ['xanadu', 'xyz', 'aardvark', 'apple', 'mix']
+# Hint: this can be done by making 2 lists and sorting each of them
+# before combining them.
+def front_x(words):
+    x=[]
+    other=[]
+    for item in words:
+        if item.startswith("x"): x.append(item)
+        else: other.append(item)
+    return sorted(x)+sorted(other)
+
+# C. sort_last
+# Given a list of non-empty tuples, return a list sorted in increasing
+# order by the last element in each tuple.
+# e.g. [(1, 7), (1, 3), (3, 4, 5), (2, 2)] yields
+# [(2, 2), (1, 3), (3, 4, 5), (1, 7)]
+# Hint: use a custom key= function to extract the last element form each tuple.
+def sort_last(tuples):
+    return sorted(tuples, key=key)
+def key(tuples):
```

Comment:
proper names for functions worth their weight in gold, try communicating intention with your identifier names.